client/packages/system/src/Item/context/useUnitVariant.tsx

```diff
@@ -1,71 +1,79 @@
-import React, { PropsWithChildren, createContext, useContext } from 'react';
-import { useItemVariants } from '../api';
+import { useUnitVariantList } from '../api';
+import { create } from 'zustand';
+import isEqual from 'lodash/isEqual';
+import { UnitVariantNode } from '@common/types';
 import { NumUtils } from '@common/utils';
+import { useEffect } from 'react';
 
-export const useUnitVariant = (itemId?: string) => {
-  const { data } = useItemVariants();
-  const item = data?.find(item => item.id === itemId);
-
-  const asPackUnit = (packSize: number) => {
-    return (
-      item?.variants.find(variant => variant.packSize === packSize)
-        ?.shortName || `${item?.unit} x ${packSize}`
-    );
-  };
-
-  // calculates SOH, AMC, MOS, Target, Suggested and Requested based on default or user selected variant
-  const numberOfPacksFromQuantity = (
-    totalQuantity: number,
-    packSize: number
-  ) => {
-    return NumUtils.round(totalQuantity / packSize, 2);
-  };
-
-  const options = item?.variants ?? [];
-
-  // const quantityFromNumberOfPacks - reverse of above
-
-  // set in app data
-  const setDefaultOption = () => {};
-
-  const defaultOption = () => {
-    // TODO: search app data first to see if user has preferred preference
-    return item?.variants.find(variant =>
-      variant.longName.includes(item?.unit)
-    );
+interface UnitState {
+  // From back end
+  items: {
+    [itemId: string]: UnitVariantNode;
   };
+  // Should be called on startup when fetching multi unit variants
+  setItems: (newItems: UnitVariantNode[]) => void;
+}
 
+const useUnitStore = create<UnitState>(set => {
```

Comment:
I thought zustand was very appropriate here, rather then context, especially with needed to select state just for particular item (and only re render the cell for that item when say new option is picked from drop down list)